src/openrct2/actions/IGameAction.h

```diff
@@ -68,12 +69,14 @@ struct GameActionResult
     rct_xyz32       Position = { 0 };
     money32         Cost = 0;
     uint16          ExpenditureType = 0;
-    void *          Tag = nullptr;
+    uint64          Results[4];
```

Comment:
This is my poor attempt to add storage per custom defined GameActionResult object, the object will need to remap result variables onto the storage currently. I have to yet find a better way to support defining custom results with copy semantics. One way would be probably to make some sort of virtual Copy function but then each custom result object is required to implement that.